prow/pjutil/filter_test.go

```diff
@@ -78,6 +78,12 @@ func TestTestAllFilter(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
+			for i := range testCase.presubmits {
```

Comment:
This is a bit smelly, why do we need it?

Reply:
Because I added a check in https://github.com/kubernetes/test-infra/pull/12836/commits/42a58e22f7fde838073dbb4fbec717591810c42a that `Trigger` is not an emtpy string, because that will result in a `matchAll` regex

Reply:
Why do we need to change that?

Reply:
`SetPresubmitRegexes` does validation, the empty trigger case happens to pass that validation by concidence because it results in a regex that matches everything. I figured I'd complete its validation by adding that check.

I can move the commit into a dedicated PR if you want.